mlir/lib/Dialect/GPU/Transforms/SubgroupReduceLowering.cpp

```diff
@@ -370,25 +371,27 @@ Value createSubgroupDPPReduction(OpBuilder &b, Location loc, Value input,
                                  gpu::AllReduceOperation mode,
                                  const ClusterInfo &ci,
                                  amdgpu::Chipset chipset) {
-  Value result = input;
   Value dppResult;
+  Value result = input;
   const int allRows = 0xf;
   const int allBanks = 0xf;
   const bool boundCtrl = true;
+  Value lane31 = b.create<LLVM::ConstantOp>(loc, b.getI32Type(), 31);
```

Comment:
Don't create a `llvm.constant` here - make an `arith.constant` and let further lowerings handle it